aten/src/ATen/ThreadLocalState.h

```diff
@@ -7,6 +7,29 @@
 
 namespace at {
 
+// Enumerating some of the thread local
+// settings we're interested in propagating
+// across threads and network
+enum class ThreadLocalSetting : uint8_t {
+  // Note: grad_mode is not automatically propagated from
+  // the forward pass into the backward pass threads
+  GRAD_MODE = 0,
+  // Whether or not to execute record functions
+  RECORD_FUNCTION,
+  // Profiler enabled/disabled
+  PROFILER,
+  NUM_SETTINGS, // must be the last in the list
```

Comment:
you can be nice and keep a few PRIVATE_USE_1/2/3 to allow other libraries to plug in